internal/scan/runtime.go

```diff
@@ -144,6 +144,7 @@ func fixConversion(node *yaml.RNode) (*yaml.RNode, error) {
 func newObjectForKind(obj runtime.Object) runtime.Object {
 	if gvks, _, _ := Scheme.ObjectKinds(obj); len(gvks) > 0 {
 		if typed, err := Scheme.New(gvks[0]); err == nil {
+			typed.GetObjectKind().SetGroupVersionKind(gvks[0])
```

Comment:
Export was doing an extra Unstructured conversion to get this information, it's easier to just add while we have it.